File: python/283_Move_Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        # y point to the position that first zero elem may exist
        y = 0
        for x in range(len(nums)):
            if nums[x]:
                nums[x], nums[y] = nums[y], nums[x]
                y += 1


def test():
    s = Solution()
    temp = [0, 1, 0, 3, 12]
    s.moveZeroes(temp)
    print temp


if __name__ == '__main__':
    test()

File: C++/283_Move_Zeroes.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
	void moveZeroes(vector<int>& nums) {
		int y = 0;
		for (int i = 0; i < nums.size(); i++){
			if (nums[i]){
				int temp = nums[i];
				nums[i] = nums[y];
				nums[y] = temp;
				y++;
			}
		}
	}
};

int main(){
	Solution s;
	int temp_arr[5] = { 0, 1, 0, 3, 12 };
	vector<int> temp(temp_arr, temp_arr + 5);
	s.moveZeroes(temp);
	cout << "[";
	for (int i = 0; i < temp.size(); i++){
		cout << temp[i];
		if (i == temp.size() - 1)
			break;
		cout << ",";
	}
	cout << "]";
	return 0;
}